components/Ripple/src/MDCRippleTouchController.m

```diff
@@ -38,70 +59,107 @@ - (instancetype)init {
     _gestureRecognizer.delaysTouchesEnded = NO;
 
     _shouldProcessRippleWithScrollViewGestures = YES;
-    _rippleView = [[MDCRippleView alloc] init];
   }
   return self;
 }
 
+- (void)dealloc {
+  [_view removeGestureRecognizer:_gestureRecognizer];
+  _gestureRecognizer.delegate = nil;
+}
+
+- (MDCRippleView *)rippleView {
+  if (_rippleView == nil) {
+    _rippleView = [[MDCRippleView alloc] init];
+  }
+  return _rippleView;
+}
+
+- (void)setDelegate:(id<MDCRippleTouchControllerDelegate>)delegate {
+  _delegate = delegate;
+
+  _delegateFlags.rippleTouchControllerShouldProcessRippleTouchesAtTouchLocation =
+      [_delegate respondsToSelector:@selector(rippleTouchController:
```

Comment:
Tip: Since `respondsToSelector` is a runtime method, does it make sense to cache the values at assignment rather than calling it as-needed?  Maybe the caching behavior should be controlled via the new delay flag?

I know it's an edge case, but at a minimum it looks like the property should be updated to document that the delegate's behavior is cached at assignment rather than inspected at runtime. Since runtime inspection is typical for `@optional` protocol methods, I think it's helpful to point this out.  

For example, if the immediate delegate actually offloads delegate to other objects (it acts as a multiplexer), then it might not have configured all of its internal state when it's assigned as the delegate for this view. Later (before the view is in the view hierarchy) it has completed its own configuration (perhaps in `viewDidLoad` of a view controller) and will respond correctly.

Reply:
We are using this pattern across the Texture code base quit a bit: https://github.com/texturegroup/texture and I didn't hear any problems with it yet. I let it is for now, that said if @yarneo or @jverkoey think are for changing it, I'm happy to do this.

Reply:
I expect the edge case outlined here to be pretty rare, but in the interest of transparency and elevating this pattern a snippet of docs can go a long way. Consider adding something to the effect of:

> The delegate's behavior - in terms of which optional methods are deemed to be
> implemented - is cached at assignment rather than inspected at runtime

Reply:
Sounds great! Added.